File: Backend/models/order.model.js
const mongoose = require('mongoose');

const orderSchema = new mongoose.Schema({
    product: { type: mongoose.Schema.Types.ObjectId, ref: 'Product', required: true },
    customerName: { type: String, required: true },
    quantity: { type: Number, default: 0 },
    type: {
        type: String,
        enum: ['in delivery', 'delivered', 'pending','return broken','return scam'],
        default: 'in delivery'
    }

}, { timestamps: true });

module.exports = mongoose.model('Order', orderSchema);

File: flutter_qr_scanner_app/lib/models/order_model.dart
class Order {
  final String id;
  final String customerName;
  final int quantity;
  final String type;
  final Map<String, dynamic>? product;

  Order({
    required this.id,
    required this.customerName,
    required this.quantity,
    required this.type,
    this.product,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['_id'],
      customerName: json['customerName'],
      quantity: json['quantity'] ?? 0,
      type: json['type'] ?? 'in delivery',
      product: json['product'] is Map<String, dynamic> ? json['product'] : null,
    );
  }
}
